tests/0102-static_group_rebalance.c

```diff
@@ -0,0 +1,217 @@
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "test.h"
+
+/* Typical include path would be <librdkafka/rdkafka.h>, but this program
+ * is built from within the librdkafka source tree and thus differs. */
+#include "rdkafka.h" /* for Kafka driver */
+
+/**
+ * KafkaConsumer static group membership tests
+ *
+ * Runs two consumers subscribing to  a topic simulating various
+ * reblance scenerios. 
+ */
+
+typedef struct _consumer_s {
+        rd_kafka_t *rk;
+        int assign_cnt;
+        int max_rebalance_cnt;
+} _consumer_t;
+
+rd_list_t worker_list;
+
+static void do_consume (_consumer_t *cons, int timeout_s) {
+        rd_kafka_message_t *rkm;
+
+        rkm = rd_kafka_consumer_poll(cons->rk, 100+(timeout_s*1000));
+        if (!rkm)
+                return;
+
+        if(rkm->err)
+                TEST_SAY(
+                    "%s consumer error: %s",
+                    rd_kafka_name(cons->rk),
+                    rd_kafka_message_errstr(rkm));
+
+        rd_kafka_message_destroy(rkm);
+
+        if (timeout_s > 0) {
+                TEST_SAY("%s: simulate processing by sleeping for %ds\n",
+                         rd_kafka_name(cons->rk), timeout_s);
+                rd_sleep(timeout_s);
+        }
+}
+
+/* Serve op queue until we have an assignment */
+static void await_assignment (_consumer_t *rk_c) {
+        TEST_SAY("Waiting for partition assignment\n");
+        while (1) {
+                rd_kafka_topic_partition_list_t *parts = NULL;
+
+                do_consume(rk_c, 1/*1s*/);
+
+                if (rd_kafka_assignment(rk_c->rk, &parts) !=
+                    RD_KAFKA_RESP_ERR_NO_ERROR ||
+                    !parts || parts->cnt == 0) {
+                        if (parts)
+                                rd_kafka_topic_partition_list_destroy(parts);
+                        continue;
+                }
+
+                TEST_SAY("%s got assignment of %d partition(s)\n",
+                         rd_kafka_name(rk_c->rk), parts->cnt);
+                rd_kafka_topic_partition_list_destroy(parts);
+                break;
+        }
+}
+
+static void rebalance_cb (rd_kafka_t *rk,
+                          rd_kafka_resp_err_t err,
+                          rd_kafka_topic_partition_list_t *parts,
+                          void *opaque) {
+
+        _consumer_t *rk_c = opaque;
```

Comment:
This is not a good variable name, `rk_*` is typically an rd_kafka_t instance:
`git grep 'rd_kafka_t \*rk_.*;'`

Try to use the same variable name for the same thing across the code, so in this case go with `c`